clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -3307,6 +3341,58 @@ Sema::DeduceTemplateArguments(VarTemplatePartialSpecializationDecl *Partial,
   return ::DeduceTemplateArguments(*this, Partial, TemplateArgs, Info);
 }
 
+TemplateDeductionResult
+Sema::DeduceTemplateArgumentsFromType(TemplateDecl *TD, QualType FromType,
```

Comment:
We have a lot of `DeduceTemplateArguments` function so I think we want to add a comment here describing what that function does / when it is used.

Reply:
This function is public (the documentation and comments are in `Sema.h` header).